src/main/java/castalia/Dist.java

```diff
@@ -0,0 +1,381 @@
+
+
+    public static double pgamma(double x, double p, double scale)
+    { double pn1, pn2, pn3, pn4, pn5, pn6, arg, c, rn, a, b, an;
+        double sum;
+        double   oflo = 1.0e+37, xbig = 1.0e+8,
+                plimit = 1000.0, elimit = -88.0;
+
+	/* check that we have valid values for x and p */
+
+        if (p <= 0.0 || scale <= 0.0) return Double.NaN;
+        x /= scale;
+        if (x <= 0.0) return 0.0;
+
+     /* use a normal approximation if p > plimit */
+
+        if (p > plimit)
+        { pn1 = 3.0 * Math.sqrt(p) * Math.pow((x / p),
+                (1.0 / 3.0) + 1.0 / (9.0 * p) - 1.0);
+            return pnorms(pn1);
+        }
+     /* if x is extremely large compared to p then return 1 */
+        if (x > xbig) return 1.0;
+        if (x <= 1.0 || x < p)
+        { /* use pearson's series expansion. */
+            arg = p * Math.log(x) - x - lgamma(p + 1.0);
+            c = 1.0; sum = 1.0; a = p;
+            do { a += 1.0; c *= x / a; sum += c; } while (c > DBL_EPSILON);
+            arg += Math.log(sum); sum = 0.0;
+            if (arg >= elimit) sum = Math.exp(arg);
+        }
+        else
+        { /* use a continued fraction expansion */
+            arg = p * Math.log(x) - x - lgamma(p);
+            a = 1.0 - p; b = a + x + 1.0; c = 0.0;
+            pn1 = 1.0; pn2 = x; pn3 = x + 1.0; pn4 = x * b;
+            sum = pn3 / pn4;
+            for (;;)
+            { a += 1.0; b += 2.0; c += + 1.0; an = a * c;
+                pn5 = b * pn3 - an * pn1; pn6 = b * pn4 - an * pn2;
+                if (Math.abs(pn6) > 0.0)
+                { rn = pn5 / pn6;
+                    if (Math.abs(sum - rn) <=
+                            Math.min(DBL_EPSILON, DBL_EPSILON * rn)) break;
+                    sum = rn;
+                }
+                pn1 = pn3; pn2 = pn4; pn3 = pn5; pn4 = pn6;
+                if (Math.abs(pn5) >= oflo)
+                { /* re-scale terms in continued fraction if terms are large */
+                    pn1 /= oflo; pn2 /= oflo; pn3 /= oflo; pn4 /= oflo;
+                }
+            }
+            arg += Math.log(sum); sum = 1.0;
+            if (arg >= elimit) sum = 1.0 - Math.exp(arg);
+        }
+        return sum;
+    }
+
+    /** gamma quantile function */
+    public static double qgamma(double p, double alpha, double scale)
+    { /* Based on Algorithm AS 91 */
+        int MAXIT= 20;
+        double C1=0.01, C2=0.222222, C3=0.32, C4=0.4, C5=1.24,
+                C6=2.2, C7=4.67, C8=6.66, C9=6.73, C10=13.32,
+                C11=60.0, C12=70.0, C13=84.0, C14=105.0, C15=120.0,
+                C16=127.0, C17=140.0, C18=1175.0, C19=210.0, C20=252.0,
+                C21=2264.0, C22=294.0, C23=346.0, C24=420.0, C25=462.0,
+                C26=606.0, C27=672.0, C28=707.0, C29=735.0, C30=889.0,
+                C31=932.0, C32=966.0, C33=1141.0, C34=1182.0, C35=1278.0,
+                C36=1740.0, C37=2520.0, C38=5040.0;
+
+        double a, b, c, ch, g, p1, v;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://www.codacy.com/app/scala-academy/Castalia/file/1968742700/issues/source?bid=2998213&fileBranchId=3112713#l323)